BusinessLayer/SongControl/SongControl.cpp

```diff
@@ -16,24 +20,42 @@ SongControl::~SongControl()
 QString SongControl::nextSong()
 {
     currentSongIndex_ = (currentSongIndex_ + 1) % files_.size();
+    playOrder_.prepend(currentSongIndex_);
+    playOrderIndex_ = 0;
     return files_[currentSongIndex_];
 }
 
 QString SongControl::previousSong()
 {
-    currentSongIndex_ -= 1;
-    if (currentSongIndex_ < 0)
+
+    if (playOrderIndex_ < 9)
     {
-        currentSongIndex_ = files_.size() - 1;
+        playOrderIndex_ += 1;
     }
-    return files_[currentSongIndex_];
+
+    if (playOrder_[playOrderIndex_] == -1)
```

Comment:
As mentioned in person, lets have it so shuffling and going to a previous song lets plays another random song so we can keep reliable behaviour when not shuffling